library/x509write_csr.c

```diff
@@ -230,6 +226,11 @@ static int x509write_csr_der_internal(mbedtls_x509write_csr *ctx,
         return MBEDTLS_ERR_X509_INVALID_ALG;
     }
 
+    if ((ret = mbedtls_pk_sign_ext((mbedtls_pk_sigalg_t) pk_alg, ctx->key, ctx->md_alg, hash, 0,
```

Comment:
In this function, `mbedtls_pk_type_t pk_alg` is always set to either `MBEDTLS_PK_RSA` or `MBEDTLS_PK_ECDSA`, without going through auxiliary functions. Please change it to `mbedtls_pk_sigalg_t` and remove the casts.